### ISSUE TITLE
Search by location is messy

### ISSUE THREAD

**Issue by hermesespinola** [immich-app/immich#8940]
### The bug

On the Search tab, use "Search Location", when selecting the filters top to bottom, it correctly constraints the following fields (i.e.: selecting "Italy" as a country limits the State and City select fields to states and cities in Italy. If you select "City" before "state" and "country" it won't limit the previous fields, making it possible to create invalid location filters:

![location-search-bug](https://github.com/immich-app/immich/assets/6611754/5dc93066-f2ac-4d5e-a1c2-ff17e2a48119)

### The OS that Immich Server is running on

Linux raspberrypi 6.1.0-rpi7-rpi-2712

### Version of Immich Server

v1.101.0

### Version of Immich Mobile App

v1.101.0 build.147

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
I don't think this is relevant
```

### Your .env content

```Shell
I don't think this is relevant
```

### Reproduction steps

```bash
1. Go to search tab
2. Click on the "search your photos" bar
3. Click on the "Location" chip
4. Select a city, then a state, then a country
```

### Additional information

I noticed on web what it does is it clears the next fields to whatever you just set (i.e.: if you select "Rome" in the City field, then select "Spain" in the Country field it will clear "Rome from the search). I think it would be nice to have this same behavior on mobile.

**Comment by ciccioska** [comment]
When you select Country> State> City and change Country it should clear State and City, if we shift State leaving the previous Country selected it should clear City as the web app works.

**Comment by Tushar-Harsora** [comment]
Attempted Fix in #9261
